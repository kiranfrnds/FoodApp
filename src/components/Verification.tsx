import ReplayIcon from "@mui/icons-material/Replay";
import { log } from "console";
import React, { useEffect, useState } from "react";
import OtpInput from "react-otp-input";
import { useNavigate, useParams } from "react-router-dom";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

const Verification = () => {
  const { number } = useParams();

  const [password, setPassword] = useState<any>("");
  const [otp, setOtp] = useState<any>("");
  const [counter, setCounter] = useState<any>(59);

  const navigate = useNavigate();

  useEffect(() => {
    const timer: any =
      counter > 0 && setInterval(() => setCounter(counter - 1), 1000);
    return () => clearInterval(timer);
  }, [counter]);

  const handleChange = (event: any) => {
    setPassword(event);
    setOtp(event);
  };

  const onClickHandle = async (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    const Data: any = {
      username: number,
      password: password,
    };
    await fetch(
      "https://extended-retail-app.herokuapp.com/api/customer/login",
      {
        method: "POST",
        body: JSON.stringify(Data),
        headers: {
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      }
    )
      .then((resp) => resp.json())
      .then((respo) => {
        if (respo.status === 200) {
          sessionStorage.setItem("logindata", JSON.stringify(respo));
          toast("Login Successful");
          navigate("/product");
        } else {
          toast("Please Enter valid OTP ");
        }
      })
      .catch((err) => {
        console.log("error", err);
      });
  };

  return (
    <div className="bg-container">
      <div className="bottom-bg">
        <img
          src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702958/logo_1_seaocx.png"
          alt="food-app"
          className="food-logo"
        />
        <h4 className="verify-head">Verification</h4>
        <p className="otp-desc">Enter the otp sent to +91 {number}</p>
        <div>
          <OtpInput
            inputStyle={{
              width: "60px",
              height: "50px",
              borderRadius: "5px",
              border: "1px solid #ccc",
              margin: "0 5px",
              fontSize: "20px",
            }}
            value={otp}
            onChange={handleChange}
            numInputs={4}
          />
        </div>
        <p className="sec">sec {counter}</p>
        <div className="tog">
          <h4 className="resend">Resend OTP</h4>
          <ReplayIcon className="repay" />
        </div>
        <button
          className={password.length < 4 ? "verify" : "not"}
          onClick={onClickHandle}
          disabled={password.length < 4}
        >
          Verification
        </button>
      </div>
      <div>
        <img
          src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652713695/Untitled-1_5_jthzoz.png"
          alt="otp"
          className="order-icon"
        />
      </div>
      <ToastContainer />
    </div>
  );
};

export default Verification;
