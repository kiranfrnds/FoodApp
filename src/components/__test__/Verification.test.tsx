import { fireEvent, render, screen } from "@testing-library/react";
import Verification from "../Verification";
import { BrowserRouter as Router } from "react-router-dom";

async function GetOtp(number: any, password: any) {
  try {
    const result = await fetch(
      `https://extended-retail-app.herokuapp.com/api/customer/login`,
      {
        method: "POST",
        body: JSON.stringify({ username: number, password: password }),
        headers: {
          "Content-Type": "application/json",
          Accept: "application/json",
        },
      }
    );
    const data = await result.json();
    if (data.status === 200) {
      return data.message;
    }
  } catch (err) {
    return null;
  }
}

describe("Verification", () => {
  beforeAll(() => {
    jest.spyOn(window, "alert").mockImplementationOnce(() => {});
    jest.spyOn(console, "error").mockImplementationOnce(() => {});
  });

  it("snapshot", () => {
    const { asFragment } = render(
      <Router>
        <Verification />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  //   it("renders input field", () => {
  //     render(
  //       <Router>
  //         <Verification />
  //       </Router>
  //     );
  //     expect(screen.getByTestId("otp-input")).toBeInTheDocument();
  //   });

  it("renders resendOtp button", () => {
    render(
      <Router>
        <Verification />
      </Router>
    );
    expect(screen.getByText("Resend OTP")).toBeInTheDocument();
  });
  it("enable submit button after entering 4 digit otp", () => {
    render(
      <Router>
        <Verification />
      </Router>
    );
    fireEvent.change(screen.getByTestId("otp-input-0"), {
      target: { value: "1" },
    });
    fireEvent.change(screen.getByTestId("otp-input-1"), {
      target: { value: "2" },
    });
    fireEvent.change(screen.getByTestId("otp-input-2"), {
      target: { value: "3" },
    });
    fireEvent.change(screen.getByTestId("otp-input-3"), {
      target: { value: "4" },
    });
    const button = screen.getByRole("button");
    expect(button).toBeDefined();
  });
  it("goto prodcut page after clicking on submit button", async () => {
    render(
      <Router>
        <Verification />
      </Router>
    );
    fireEvent.click(screen.getByRole("button"));
    const result = await GetOtp("1234567891", "1234");
    expect(result).toBeTruthy();
  });
});
