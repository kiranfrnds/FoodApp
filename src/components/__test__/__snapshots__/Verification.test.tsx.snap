// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verification snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-container"
  >
    <div
      class="bottom-bg"
    >
      <img
        alt="food-app"
        class="food-logo"
        src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652702958/logo_1_seaocx.png"
      />
      <h4
        class="verify-head"
      >
        Verification
      </h4>
      <p
        class="otp-desc"
      >
        Enter the otp sent to +91 
      </p>
      <div>
        <div
          style="display: flex;"
        >
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Please enter verification code. Character 1"
              autocomplete="off"
              class=" "
              data-testid="otp-input-0"
              maxlength="1"
              style="width: 60px; text-align: center; height: 50px; border-radius: 5px; border: 1px solid #ccc; margin: 0px 5px; font-size: 20px;"
              type="text"
              value=""
            />
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Character 2"
              autocomplete="off"
              class=""
              data-testid="otp-input-1"
              maxlength="1"
              style="width: 60px; text-align: center; height: 50px; border-radius: 5px; border: 1px solid #ccc; margin: 0px 5px; font-size: 20px;"
              type="text"
              value=""
            />
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Character 3"
              autocomplete="off"
              class=""
              data-testid="otp-input-2"
              maxlength="1"
              style="width: 60px; text-align: center; height: 50px; border-radius: 5px; border: 1px solid #ccc; margin: 0px 5px; font-size: 20px;"
              type="text"
              value=""
            />
          </div>
          <div
            style="display: flex; align-items: center;"
          >
            <input
              aria-label="Character 4"
              autocomplete="off"
              class=""
              data-testid="otp-input-3"
              maxlength="1"
              style="width: 60px; text-align: center; height: 50px; border-radius: 5px; border: 1px solid #ccc; margin: 0px 5px; font-size: 20px;"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <p
        class="sec"
      >
        sec 59
      </p>
      <div
        class="tog"
      >
        <h4
          class="resend"
        >
          Resend OTP
        </h4>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium repay css-i4bv87-MuiSvgIcon-root"
          data-testid="ReplayIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5V1L7 6l5 5V7c3.31 0 6 2.69 6 6s-2.69 6-6 6-6-2.69-6-6H4c0 4.42 3.58 8 8 8s8-3.58 8-8-3.58-8-8-8z"
          />
        </svg>
      </div>
      <button
        class="verify"
        disabled=""
      >
        Verification
      </button>
    </div>
    <div>
      <img
        alt="otp"
        class="order-icon"
        src="https://res.cloudinary.com/dq21uo3uq/image/upload/v1652713695/Untitled-1_5_jthzoz.png"
      />
    </div>
    <div
      class="Toastify"
    />
  </div>
</DocumentFragment>
`;
